rpmio/rpmpgp.c

```diff
@@ -525,9 +525,9 @@ static int pgpPrtSigParams(pgpTag tag, uint8_t pubkey_algo, uint8_t sigtype,
     int i;
     pgpDigAlg sigalg = pgpSignatureNew(pubkey_algo);
 
-    for (i = 0; i < sigalg->mpis && p + 2 <= pend; i++) {
+    for (i = 0; i < sigalg->mpis && 2 <= pend - p; i++) {
```

Comment:
The compiler doesn't of course care, but I find this order of comparison backwards and unnatural. You don't say "is two less or equal to the remaining buffer size", you say "is the remaining buffer size at least two bytes?". Ditto with all similar changes in this and other PR's.  But when you write it as "pend - p >= 2" the point actually becomes *more* obvious than the original code, and at that point I have far less objections to these changes.

Reply:
Will fix, thanks!